### DISCUSSION TITLE
Tracks and routes concerns docs.osmand.net/docs/user/map/tracks-and-routes

### DISCUSSION THREAD

**Discussion by fixifoxy** [osmandapp/OsmAnd#20505] [General]
I am just finishing the above mentioned article. Better than ever. but still of no use.

I learn that there are different GPX-Formats:
1. one in <trkpt> syntax. Lets call it "Track"
2. one  in <rtept> syntax. Lets call it "Route"

But who will tell me
1.Where and why to use the one and why the other.
2. How to create the one and how to create the other.

Could we expect to get those questions answered in the Osmand doc?
  

Let me assume that I use both files only to be shown on the map:
1. I show the "track" file on the  map. Using a profile let us say profile A. After that I show the same GPX-file (track)  with profile x. Will this change the shown track on the map?
2. Let us assume that I show a "Route" gpx-file on the map with profile A and afterwards with profile X.  Will I have to different line on the map?

3. Now I experiment with navigation:
Under navigation I navigate at first with the "track" gpx-file and profile A an the with profile X. Should I expect that I´ve got twice the same line to navigate?

Next I navigate with the "route" gpx-file and profile A and I will get the navigation line generated with profile A. 
After that I navigate the same "route" gpx-file with profile X.
Will this change the navigation to do?

**Comment by fixifoxy** [comment]
syntax 1=trkpt
syntax 2=rtept

**Reply by fixifoxy** [reply]
Is no one able to answer these questions?

Let me assume that I use both files only to be shown on the map:

    I show the "track" file on the map. Using a profile let us say profile A. After that I show the same GPX-file (track) with profile x. Will this change the shown track on the map?

    Let us assume that I show a "Route" gpx-file on the map with profile A and afterwards with profile X. Will I have to different line on the map?

    Now I experiment with navigation:
    Under navigation I navigate at first with the "track" gpx-file and profile A an the with profile X. Should I expect that I´ve got twice the same line to navigate?

Next I navigate with the "route" gpx-file and profile A and I will get the navigation line generated with profile A.
After that I navigate the same "route" gpx-file with profile X.
Will this change the navigation to do?

**Comment by EugeneZmeuk** [comment]
Hi @fixifoxy 
Here (https://github.com/osmandapp/web/issues/436) there is an example of GPX track with trkpt (track point) and rtept (route point). There is additional info about Waypoint (tag=wpt) for the classic GPX format. 

In our [Docs > "Map"](https://osmand.net/docs/user/map/tracks-on-map#types-of-tracks-and-routes) section we write all data which you can see on the map.  Article ["Tracks and routes"](https://osmand.net/docs/user/map/tracks-on-map#types-of-tracks-and-routes) is an article about GPX tracks on the map and Routes on the map.
But Routes is a not GPX track on the first iteration, but you can [generate GPX track](https://osmand.net/docs/user/map/tracks-on-map#save-as-a-track) from chosen Route on the map. [Route on the map](https://osmand.net/docs/user/map/vector-maps/#routes) is a predefined path, which is a sequence of roads and turns to be followed to reach a certain destination point. In the OsmAnd app, you can mark routes on the map related to different outdoor activities using data from [OpenStreetMap](https://wiki.openstreetmap.org/wiki/Relation:route).

**Now, about GPX:**
rtept - Route Point tag for GPX format is start-finish points for any constructor of GPX track. It means that you create a track you choose two points on the map (maybe more) and between these points a constructor builds line. For this case GPX has two syntax trkpt and rtept. where rtept is start-finish points, trkpt is all points between start-finish points (route). For [OsmAnd track planner](https://osmand.net/docs/user/plan-route/create-route) all info about was saved on GPX what Profile was used for segment (it can be two and more profiles). And when you open this GPX on OsmAnd web or OsmAnd app (and choose "Plan a route"), you can change Profile for segment. BUT if to speak about GPX track from others constructors you will have this case https://github.com/osmandapp/web/issues/436.

**Answer to your question** - NO. "Navigation by track" will be only by line your track (trkpt points). BUT -  Navigation line to Start point or nearest point of the track will depend on your Profile.

**Comment by mariush444** [comment]
https://osmand.net/docs/user/map/tracks-on-map#types-of-tracks-and-routes

doesn't work

**Reply by EugeneZmeuk** [reply]
https://osmand.net/docs/user/map/tracks/#types-of-tracks
https://osmand.net/docs/user/map/routes#type-of-routes

**Reply by fixifoxy** [reply]
Files of marius444 not available.
Answers of Zmeuk of no use.

So once again let us differentiate:
We have two kinds of GPX_files: 
1  syntax form  trkpt. And to be clear, I will call it a track. As other apps do.
2. syntax form rtept. And to be clear, I will call it a route. As other apps do.
I hope that is simple enough.

And there are apps offering the transfer from one syntax to he other or lets say from track to route.
I am interested  to understand what is the difference in the app-process when having the one or the other GPX-syntax.

**Reply by mariush444** [reply]
What I understand from their docs... track is point by point (every point is included, every 10 or 50 meters point is recorded mentioned)
so you can follow track with any profile (without any calculation)
but a route (the simplest version) is just start and end point with info about car/pedestrian/... profile.
It means (it seems) that route is calculate every time when you start it.
This is what I understnd the difference.

**Reply by EugeneZmeuk** [reply]
@fixifoxy 

"1.Where and why to use the one and why the other.''
'''2. How to create the one and how to create the other.''

- [x] 1. 
**"trkpt" Where** 
any trip recording or saving  "rtept" (A >B) with complete route.
**"trkpt" Why**
It is a simplified standard for all applications, Web tools and etc. It means that GPX format with ''trkpt'' will be opened everywhere
![image](https://github.com/user-attachments/assets/e3e186d4-ed2c-4783-8688-e8728b332484)
For understanding, this is a set of points with coordinates (minimum) and other data, which in many applications are linked together graphically, resulting in a solid route line.
<img width="319" alt="image" src="https://github.com/user-attachments/assets/bd30d89a-2396-48a9-85af-eadc6233997c">


**"rtept" Why**
Apps have own tools for creating (planning) route trip. You create this trip from one point to another with any Profile. This format saving your chosen points on the map with data - coordinates, profile... This is necessary to be able to format this planning GPX track in the "Plan a route" tool. That is, you can open a file already made in the "Plan a route" tool and edit it, change profiles for route sections, etc.
**"rtept" Where**
For GPX track which were created on any Planning trip tool (["Plan a route" tool in OsmAnd](https://osmand.net/docs/user/plan-route/create-route))
![image](https://github.com/user-attachments/assets/d5dece63-e0ec-4631-95a0-52ac8bb650cd)
<img width="315" alt="image" src="https://github.com/user-attachments/assets/b0bf6b4c-6f25-4bd9-92cd-00ef854a4bd3">

- [x] 2. for this question I answer only for OsmAnd app:
''trkpt'' - saving your trip by ["Trip recording"](https://osmand.net/docs/user/plugins/trip-recording#recorded-gpx-file).
''trkpt''- saving a route with ["Plan a route" tool](https://osmand.net/docs/user/plan-route/create-route#save-as-new-track). You have selected two points for the "Plan a route". The route between these points is built and written in the "trkpt" format to GPX.

"rtept" - saving a route with ["Plan a route" tool](https://osmand.net/docs/user/plan-route/create-route#save-as-new-track). This format saving your chosen points on the map with data - coordinates, profile... This is necessary to be able to format this planning GPX track in the "Plan a route" tool. That is, you can open a file already made in the "Plan a route" tool and edit it, change profiles for route sections, etc.
![image](https://github.com/user-attachments/assets/f2947cc6-0dd3-40ae-8d38-c5f1c1b90a98)


**NOTE:**
Planners in different applications save two formats (trkpt and rtept). When we open a created track in application A, [such problems](https://github.com/osmandapp/web/issues/436) arise in application B.
Therefore, if you have planned a route and will use it not in OSMAND, then you need to save the track as [a simplified track](https://osmand.net/docs/user/plan-route/create-route#save-as-new-track).
A simplified track contains only "trkpt" points.

**Comment by fixifoxy** [comment]
as always with osmand. Many lines. No Answer.
